src/System.Net.Security/src/System/Net/ContextFlagsAdapterPal.Unix.cs

```diff
@@ -46,9 +58,21 @@ internal static ContextFlagsPal GetContextFlagsPalFromInterop(Interop.NetSecurit
             return flags;
         }
 
-        internal static Interop.NetSecurityNative.GssFlags GetInteropFromContextFlagsPal(ContextFlagsPal flags)
+        internal static Interop.NetSecurityNative.GssFlags GetInteropFromContextFlagsPal(ContextFlagsPal flags, bool isServer)
         {
             Interop.NetSecurityNative.GssFlags gssFlags = (Interop.NetSecurityNative.GssFlags)0;
+
+            // GSS_C_INTEG_FLAG is set if either AcceptIntegrity (used by server) or InitIntegrity (used by client) is set
+            if ((flags & ContextFlagsPal.AcceptIntegrity) != 0  && isServer)
+            {
+                gssFlags |= Interop.NetSecurityNative.GssFlags.GSS_C_INTEG_FLAG;
+            }
+
+            if ((flags & ContextFlagsPal.InitIntegrity) != 0 && !isServer)
+            {
+                gssFlags |= Interop.NetSecurityNative.GssFlags.GSS_C_INTEG_FLAG;
+            }
```

Comment:
You might change this to:
```C#
if (isServer)
{
    ...
}
else
{
    ...
}
```
to help make it clear that each if block is only relevant to one of the two cases.